llvm/lib/Target/PowerPC/PPCAsmPrinter.cpp

```diff
@@ -2058,6 +2089,10 @@ void PPCAIXAsmPrinter::emitLinkage(const GlobalValue *GV,
     }
   }
 
+  // Do not emit the _$TLSML symbol.
+  if (GVSym->getName() == "_Renamed..5f24__TLSML[TC]")
```

Comment:
Maybe a silly question, but will this always contain the `5f24`?

Reply:
I think `MCContext::createXCOFFSymbolImpl` is the logic that replaced the invalid characters. According to ASCII table, `_$` maps to `5f24`.

Reply:
I think `MCContext::createXCOFFSymbolImpl` is the logic that replaced the invalid characters. According to ASCII table, `_$` maps to `5f24`...

Reply:
This check (and other checks in this PR) does a string comparison, which doesn't seem efficient.  Is there a way to set a flag on the symbol to avoid the string operations? Or are you out of flag bits?

Reply:
Thank you! I will put some flag check before the string compare.